File: ruby-impl/Sample.rb
require_relative 'hackathonlib/Client'
require_relative 'hackathonlib/Connection'
require_relative 'hackathonlib/Network'

class Sample < Client
  def update_scene(scene)

  end

  def get_command
    nil
  end

  def get_name
    'Sample'
  end

  def get_speed
    nil
  end

  def get_rotation
    nil
  end

  def do_ping
    true
  end
end

if __FILE__ == $0
  # Define hostname and port.
  hostname = 'localhost'
  port = 25555

  # Open up a tcp connection.
  client = Network.connect(hostname, port)
  connection = Connection.new(client, Sample.new, -1)

  # Start the client loop.
  connection.start_client_loop

  # Join the connection since it's callback based.
  connection.join
end

File: csharp-impl/HackathonSample/Sample.cs
using HackathonLib;
using System.Collections.Generic;

namespace HackathonSample
{
    class Sample : Client
    {
        static void Main()
        {
            // Define hostname and port.
            const string hostname = "localhost";
            const int port = 25555;

            // Open up a tcp connection.
            var client = Network.Connect(hostname, port);
            var connection = new Connection(client, new Sample(), -1, true);

            // Join the connection since it's callback based.
            connection.Join();
        }

        override public void UpdateScene(List<GameObject> scene)
        {
            // TODO put code here
        }

        override public string GetCommand()
        {
            // TODO put code here
            return null;
        }

        override public string GetName()
        {
            // TODO put code here
            return "C-Sharp Sample";
        }

        override public double? GetSpeed()
        {
            // TODO put code here
            return null;
        }

        override public double? GetRotation()
        {
            // TODO put code here
            return null;
        }

        override public bool DoPing()
        {
            // TODO put code here
            return true;
        }
    }
}
